### DISCUSSION TITLE
branches' trace

### DISCUSSION THREAD

**Discussion by CorradoLanera** [ropensci-books/targets#76] [Help]
Is there a friendly way for a given branch in a pipeline involving multiple subsequent patterns with a dynamic branch to have the list of all its parent "branches"? E.g., I read a CSV; next, a pattern branching over its record for preprocessing them in parallel; next, a pattern with their analyses; next, a pattern with some statistics on the analyses; and a final plot of the statistics altogether. If a branch in the statistics pattern throws an error, how to view the chain of branches: record_branch - preprocessing_branch - analyses_branch - statistics_branch leading to the error? Discover the name of the errored branch it's quite easy thanks to `tar_meta`, but how to go back?

**Comment by wlandau** [comment]
There are esoteric functions listed in https://docs.ropensci.org/targets/reference/index.html#branching, but my best advice is to ship descriptive metadata with each dynamic branch you create. Example: https://books.ropensci.org/targets/dynamic.html#provenance
